apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/PreAdjFissClaimTransformerV2.java

```diff
@@ -0,0 +1,230 @@
+   */
+  @Trace
+  static Claim transform(MetricRegistry metricRegistry, Object claimEntity) {
+    Timer.Context timer = metricRegistry.timer(METRIC_NAME).time();
+
+    if (!(claimEntity instanceof PreAdjFissClaim)) {
+      throw new BadCodeMonkeyException();
+    }
+
+    Claim claim = transformClaim((PreAdjFissClaim) claimEntity);
+
+    timer.stop();
+    return claim;
+  }
+
+  /**
+   * @param claimGroup the {@link PreAdjFissClaim} to transform
+   * @return a FHIR {@link Claim} resource that represents the specified {@link PreAdjFissClaim}
+   */
+  private static Claim transformClaim(PreAdjFissClaim claimGroup) {
+    Claim claim = new Claim();
+
+    claim.setId("f-" + claimGroup.getDcn());
+    claim.setMeta(new Meta().setLastUpdated(Date.from(claimGroup.getLastUpdated())));
+    claim.setContained(getContainedProvider(claimGroup));
+    claim.setIdentifier(getIdentifier(claimGroup));
+    claim.setStatus(Claim.ClaimStatus.ACTIVE);
+    claim.setType(getType());
+    claim.setUse(Claim.Use.CLAIM);
+    claim.setCreated(new Date());
+    claim.setPriority(getPriority());
+    claim.setTotal(getTotal(claimGroup));
+    claim.setProvider(new Reference("#provider-org"));
+    claim.setPatient(getPatient(claimGroup));
+    claim.setDiagnosis(getDiagnosis(claimGroup));
+    claim.setProcedure(getProcedure(claimGroup));
+
+    return claim;
+  }
+
+  private static List<Identifier> getIdentifier(PreAdjFissClaim claimGroup) {
+    return Collections.singletonList(
+        new Identifier()
+            .setType(
+                new CodeableConcept(
+                    new Coding(
+                        "http://hl7.org/fhir/us/carin-bb/CodeSystem/C4BBIdentifierType",
+                        "uc",
+                        "Unique Claim ID")))
+            .setSystem("https://dcgeo.cms.gov/resources/variables/dcn")
+            .setValue(claimGroup.getDcn()));
+  }
+
+  private static CodeableConcept getType() {
+    return new CodeableConcept()
+        .setCoding(
+            Arrays.asList(
+                new Coding("https://dcgeo.cms.gov/resources/codesystem/rda-type", "FISS", null),
+                new Coding(
+                    "http://terminology.hl7.org/CodeSystem/claim-type",
+                    "institutional",
+                    "Institutional")));
+  }
+
+  private static CodeableConcept getPriority() {
+    return new CodeableConcept(
+        new Coding("http://hl7.org/fhir/ValueSet/process-priority", "normal", "Normal"));
+  }
+
+  private static Money getTotal(PreAdjFissClaim claimGroup) {
+    Money total = new Money();
+
+    total.setValue(claimGroup.getTotalChargeAmount());
+    total.setCurrency("USD");
+
+    return total;
+  }
+
+  private static List<Resource> getContainedProvider(PreAdjFissClaim claimGroup) {
+    return Collections.singletonList(
+        new Organization()
+            .setIdentifier(
+                Arrays.asList(
```

Comment:
Style suggestion: I prefer to use`ImmutableList.of()` for these sorts of throw away lists rather than a mix of `Collections.singletonList ()` and `Arrays.asList()`.  It requires fewer imports, it's more consistent, the resulting list is clearly and always immutable (vs `Collections.singletonList()` which is but `Arrays.asList()` only sort of is), and the `ImmutableList.of()` method is overloaded with different numbers of parameters so the JVM doesn't have to create any throw away intermediate arrays for a variadic parameter.

Reply:
It's a good suggestion.  I've just naturally been prone to Collections and Arrays because they are part of the JDK.  I'm not sure about the efficiency aspect of it, `Arrays.asList()` just directly sets the field value to the passed varargs array, where as `ImmutableList.of()` does a fair bit more than that including some validations and switching.

Reply:
ImmutableList.of() just creates an ImmutableList containing the values passed to it and returns the list.  Difference is the list is guaranteed to be immutable.  Generally immutable data is safer data and best when you don't require mutability.  Here mutability isn't required.